File: app/src/main/java/com/scheffer/erik/popularmovies/utils/ConnectionUtils.java
package com.scheffer.erik.popularmovies.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class ConnectionUtils {
    public static boolean isConnected(Context context) {
        ConnectivityManager cm =
                (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);

        if (cm != null) {
            NetworkInfo activeNetwork = cm.getActiveNetworkInfo();
            return activeNetwork != null && activeNetwork.isConnectedOrConnecting();
        }
        return false;
    }
}

File: app/src/main/java/com/scheffer/erik/popularmovies/utils/ConnectionUtils.kt
package com.scheffer.erik.popularmovies.utils

import android.content.Context
import android.net.ConnectivityManager

fun isConnected(context: Context) =
        (context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager)
                .activeNetworkInfo?.isConnectedOrConnecting ?: false
